docs/rules/no-identical-expressions.md

```diff
@@ -0,0 +1,48 @@
+# no-identical-expressions
+Using the same value on either side of a binary operator is almost always a mistake. In the case 
+of logical operators, it is either a copy/paste error and therefore a bug, or it is simply wasted 
+code, and should be simplified. In the case of bitwise operators and most binary mathematical 
+operators, having the same value on both sides of an operator yields predictable results, and 
+should be simplified.
+
+This rule ignores `*`, `+`, and `=`.
+
+## Noncompliant Code Example
+
+```javascript
+if ( a == a ) { // always true
+  doZ();
+}
+
+if ( a != a ) { // always false
```

Comment:
I wonder if we should exclude self comparison because of the core rule

Reply:
@stas-vilchik in fact it's excluded already by implementation (as such code can be used for NaN check), I will update description